mlrun/model.py

```diff
@@ -221,15 +221,26 @@ def _apply_enrichment_before_to_dict_completion(
 
     @classmethod
     def from_dict(
-        cls, struct=None, fields=None, deprecated_fields: Optional[dict] = None
+        cls,
+        struct=None,
+        fields=None,
+        deprecated_fields: Optional[dict] = None,
+        init_with_params: bool = False,
     ):
         """create an object from a python dictionary"""
         struct = {} if struct is None else struct
         deprecated_fields = deprecated_fields or {}
         fields = fields or cls._dict_fields
         if not fields:
             fields = list(inspect.signature(cls.__init__).parameters.keys())
-        new_obj = cls()
+
+        if init_with_params:
+            kwargs = {field: struct.get(field, None) for field in fields}
+            kwargs.pop("self", None)
+            new_obj = cls(**kwargs)
+        else:
+            new_obj = cls()
```

Comment:
Need to consider 3 main points : 
1. What happen when user define `_dict_fields` and passes `init_with_params=True` - I think you should use the signature when you initialized.  
2. What happens if `**kwargs` is part of the `cls` signature.
3. Why not to delete the fields you already use for the initialization from the struct?

Reply:
Do we really need to introduce the new `init_with_params` parameter? I mean, why not do as we do below, and check if `struct` exist in order to decide what is needed?